chainntnfs/mempool.go

```diff
@@ -77,9 +83,20 @@ func (m *MempoolNotifier) SubscribeInput(
 	outpoint wire.OutPoint) *MempoolSpendEvent {
 
 	Log.Debugf("Subscribing mempool event for input %s", outpoint)
+
+	// Get the current subscribers for this input or create a new one.
+	clients := &lnutils.SyncMap[uint64, *MempoolSpendEvent]{}
```

Comment:
Is this upfront declaration actually used? We only use the variable in a singel code path below.

Reply:
so `clients` is used in `LoadOrStore` below. When the targeted input is not found, this `clients` will be saved into `subscribedInputs` as an initialization of the map.

Reply:
Ah yeah it basically lets you skip the nested statement of `!ok -- create map`.